### DISCUSSION TITLE
Accessing Request directly from a parametrized dependency's __call__ function.

### DISCUSSION THREAD

**Discussion by Ozanaydinn** [fastapi/fastapi#13104] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# main.py
from services.auth_service import AuthenticationChecker
from fastapi import FastAPI

app = FastAPI()

# Add API routers and authentication logic
admin_auth = AuthenticationChecker(required_role="Admin")

app.include_router(management.router,
                   prefix="/mgmt",
                   tags=["management"],
                   dependencies=[Depends(admin_auth)])

# /services/auth_service.py
from typing import Optional
from starlette.requests import Request
from fastapi import Header

class AuthenticationChecker:
    def __init__(self, required_role: Optional[str] = None):
        self.required_role = required_role

    async def __call__(self,
                       request: Request,  #This does not work
                       x_access_token: str = Header(None)) -> bool:
       # Here, I need to access the request object and do some authentication.
         
# Example management route
from pydantic import BaseModel
from fastapi import APIRouter

router = APIRouter()

class ConfigRequest(BaseModel):
    project: str
    config_type: str
    config_data: List[str]

@router.post("/set_config")
async def set_config(config_request: ConfigRequest):
   pass
```

### Description

I followed the documentation on how to create parametrized dependencies to create my `AuthenticationChecker`, and I am globally injecting an instance of it into all of my API functions under the `management` route. 

In order to access the `Request` object from the `__call__` function, I have added `request: Request` in the function signature but now I'm getting a 422 error with this explanation:
`{"detail":[{"type":"missing","loc":["query","request"],"msg":"Field required","input":null}]}`. I'm certain that the problem arises from my handling of the `Request` class but I could not find a solution for it.

Is what I'm trying to do possible? 

### Operating System

Windows

### FastAPI Version

0.115.6

### Pydantic Version

2.10.4

### Python Version

3.9.20

**Comment by sehraramiz** [comment, marked as answer]
i tried this code you provided and its working, i don't get any errors and also got the request object in the dependency call.
i think what you provided as sample code is not coordinated with your provided error message. please test this fixed single file code and let us know the result. maybe you are missing some additional code from your main project.

```python

  from typing import Optional, List
  from starlette.requests import Request
  from fastapi import Header, Depends
  from fastapi import FastAPI
  from pydantic import BaseModel
  from fastapi import APIRouter


  app = FastAPI()


  class AuthenticationChecker:
      def __init__(self, required_role: Optional[str] = None):
          self.required_role = required_role

      async def __call__(
          self, request: Request, x_access_token: str = Header(None)
      ) -> bool:
          print(request)


  router = APIRouter()


  class ConfigRequest(BaseModel):
      project: str
      config_type: str
      config_data: List[str]


  @router.post("/set_config")
  async def set_config(config_request: ConfigRequest):
      pass


  admin_auth = AuthenticationChecker(required_role="Admin")

  app.include_router(
      router,
      prefix="/mgmt",
      tags=["management"],
      dependencies=[Depends(admin_auth)],
  )

```

**Reply by YuriiMotov** [reply]
Works for me as well.
I wonder if it might be connected to OS used? I use Ubuntu

**Reply by sehraramiz** [reply]
> Works for me as well. I wonder if it might be connected to OS used? I use Ubuntu

yeah i'm on linux too.
